dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/event/RumEventMapperTest.kt

```diff
@@ -0,0 +1,186 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.domain.event
+
+import android.util.Log
+import com.datadog.android.event.EventMapper
+import com.datadog.android.log.internal.logger.LogHandler
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.datadog.android.utils.forge.Configurator
+import com.datadog.android.utils.mockSdkLogHandler
+import com.datadog.android.utils.restoreSdkLogHandler
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class RumEventMapperTest {
+    lateinit var testedRumEventMapper: RumEventMapper
+
+    @Mock
+    lateinit var mockResourceEventMapper: EventMapper<ResourceEvent>
+
+    @Mock
+    lateinit var mockActionEventMapper: EventMapper<ActionEvent>
+
+    @Mock
+    lateinit var mockErrorEventMapper: EventMapper<ErrorEvent>
+
+    @Mock
+    lateinit var mockViewEventMapper: EventMapper<ViewEvent>
+
+    @Mock
+    lateinit var mockLogHandler: LogHandler
+
+    lateinit var originalLogHandler: LogHandler
+
+    @BeforeEach
+    fun `set up`() {
+        originalLogHandler = mockSdkLogHandler(mockLogHandler)
+        testedRumEventMapper = RumEventMapper(
+            actionEventMapper = mockActionEventMapper,
+            viewEventMapper = mockViewEventMapper,
+            resourceEventMapper = mockResourceEventMapper,
+            errorEventMapper = mockErrorEventMapper
+        )
+    }
+
+    @AfterEach
+    fun `tear down`() {
+        restoreSdkLogHandler(originalLogHandler)
+    }
+
+    @Test
+    fun `M map the bundled event W map { internal mapper returns NOT NULL }`(forge: Forge) {
```

Comment:
In this year you're duplicating the switch from there implementation. I think it might be better to split this into four tests (one for each type)